### ISSUE TITLE
JSON getting into loop for some struct🐛 

### ISSUE THREAD

**Issue by DinYehoshua** [gofiber/fiber#1879]
v2.27.0

Hi,
I try to use c.JSON(data) for converting my struct into json and i'm getting into loop for some of the calls. (depends on the data on the struct)

My struct is a little complex and it had nested structs in it.

```go
package main

import "github.com/gofiber/fiber/v2"

type FCL struct {
	Details  Obj      `json:"details"`
	Policies Policies `json:"policies"`
}

type Obj struct {
	Resource map[string]interface{} `json:",inline`
}

type Policies struct {
	Policies1 []Policies1 `json:"policies1"`
	Policies2 []Policies2 `json:"policies2"`
}

type Policies1 struct {
	Name             string    `json:"name"`
	Path             string    `json:"path"`
	ModificationTime time.Time `json:"modificationTime"`
}

type Policies2 struct {
	Name             string    `json:"name"`
	Path             string    `json:"path"`
	ModificationTime time.Time `json:"modificationTime"`
}

func main() {

  data := FCL{...}
  return c.Status(200).JSON(data)
}
```

for some cases when data is with Policies2 = nil, the JSON function getting into infinity loop.
But in other cases when all fields aren't nil, everything work as expected

**Comment by ReneWerner87** [comment]
@DinYehoshua please use the latest version v2.32.0 , there are some fixes for json included in the version from 2.27.0-2.32.0

**Comment by DinYehoshua** [comment]
Update my version fixed the problem,
Thanks!
